.github/workflows/deploy.yml

```
@@ -1,18 +1,121 @@
 name: 'Deploy Application'
 
+
 on:
   workflow_call:
     inputs:
+      default-agent:
+        required: false
+        type: string
+        default: self-hosted
+      aws-region:
+        required: true
+        type: string
+      role-to-assume:
+        required: true
+        type: string
+      role-duration-seconds:
+        required: true
+        type: number
+        default: 1200
+      role-session-name:
+        required: true
+        type: string
+        default: "github-actions"
+      cluster-name:
+        required: true
+        type: string
+      app-name:
+        required: true  
+        type: string
       environment:
         required: true
         type: string
```

```suggestion
      environment:
        required: true
        type: string
        description: deployment environment (i.e. staging, production)
```

maybe we might need to change the deployment folder pattern, from `stg` to `staging` and `prd` to `production`.

Github only recognizes as "production" by using the whole name, otherwise it considers an "alternate environment".

---

Make sense